### ISSUE TITLE
Testing This Library

### ISSUE THREAD

**Issue by danielstern** [danielstern/ngAudio#144]
I've been wanting to add some tests for a while, but I'm not sure there's any means of automating the testing of if audio plays / sounds right, in various browsers.

Any ideas?

**Comment by tlenex** [comment]
Firstly don't test if audio API works! Test only **your own** code. Testing Audio API depends not only on browser, but also codecs supported by that browser, OS and other software. It should be done by API developers not you.

You can set up event listeners eg.: https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Media_events to listen if it was triggered. Some of the events may have access to useful data to test too.

**Comment by danielstern** [comment]
@tlenex Thanks for the great input! This is a good starting point for sure

**Comment by tlenex** [comment]
What I can tell more... consider using [travis](https://travis-ci.org) or other CI tool to run tests on each PR / push

**Comment by danielstern** [comment]
@tlenex yes, I think Travis makes a lot of sense for this as it is an open source friendly site. If you wanted to take the lead here, I'd be happy to set you up as a repo maintainer.

**Comment by tlenex** [comment]
I'll think about it and let you know. Currently working on other projects
